File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.java
import java.util.*;
class Solution {
    public int[] solution(int n, long left, long right) {
        int[] answer = {};
        List<Integer> ans = new ArrayList();
        
        for (long i = left+1;i<=right+1;i++){
            long x = i / n + 1;
            long y = i % n;
            
            if (y == 0) ans.add(n);
            else {
                int val = (int) Math.max(x,y);
                ans.add(val);
            }
        }
        
        
        
        return ans.stream().mapToInt(x -> x).toArray();
    }
}

File: 프로그래머스/2/87390. n＾2 배열 자르기/n＾2 배열 자르기.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(int n, long long left, long long right) {
    vector<int> answer;

    for (long long i = left; i <= right; i++) {
        int ans = max(i / n, i % n);
        answer.push_back(ans+ 1);
    }

    return answer;
}